### PR TITLE
refactor: architecture overhaul — workflow seam, projection layer, dead code removal, perf fix, E2E suite

### PR DESCRIPTION
**Pull request by seeyabye**
## Summary

Single squash commit on top of `main` (`b81ac71b`) consolidating 4 commits of architecture-refactor work: workflow seam consolidation, MovieView projection layer, dead code removal, batched-query perf fix, full-stack E2E regression suite, and a coverage boost from 88.67% → 90.46%.

## Architecture overhaul

- **Workflow seam pattern:** Replace ad-hoc task types (`ScrapeTask`, `DownloadTask`, `OrganizeTask`, `NFOTask`) with a unified `Workflow` seam. Ad-hoc legacy worker types (`JobQueue`, `ProgressTracker`, `Task`, `BatchScrapeTask`, `FieldSources`, `DisplayTitleApplier`, etc.) deleted with no dangling references.
- **Unified factory path:** Collapse `NewFromDeps`/`NewScrapeOnly` into a single `NewWorkflowFactory` path with three mode-specific accessors (`NewWorkflow`, `NewScrapeOnlyWorkflow`, `NewScanOnlyWorkflow`).
- **Extract `boundedFanOut[T,I]` skeleton** shared by `scrape_phase.go` and `apply_phase.go`; collapse duplicate `broadcastFailure` into a single shared definition.
- **`bootstrapWorkflow`** extracted shared by `Bootstrap` and `BootstrapScrapeOnly` in `commandutil`.
- **Drop pass-through seams:** Remove `JobStore`/`JobPersistencer` DB-query pass-through, `WorkflowSetter` orphan, legacy `EnsureRuntime` wrapper, and `worker_types` stub. Rename `process_*` helpers to honest names in `api/batch`.

## Projection layer

- Introduce **`MovieView`** in `internal/api/contracts/movie_view.go` separating scrape-output shape from DB row. Fully wired through `api/movie/*` and `api/batch/*` handlers; `batch_types_compat.go` provides backward-compat for legacy persisted JSON.
- Extract **`TranslationOutput`** as a distinct scrape-output type; remove redundant carrier fields on `Movie`.

## Perf

- `ListJobsWithStats`: swap per-id `CountByBatchJobID` + `CountByBatchJobIDAndRevertStatus` (2N+1 queries) for batched `CountByBatchJobIDs` + `CountRevertedByBatchJobIDs` (2 queries). Public API shape unchanged. `GetJobWithStats` correctly left on per-id path.

## Tests / tooling

- **Coverage boost:** 88.67% → 90.46% across 9 packages (`logging` 54.2% → 81.0%, plus gains in `scrape`/`worker`/`workflow`/`api`).
- **Full-stack E2E regression suite** added under `web/frontend/tests/fullstack/regressions/` (19 spec files covering auth, batch-rescrape, cache-hit, cancel-job, delete-job, events, exclude-result, field-drop, health, jobs-list/inlined-data/operations-recorded, movies-detail/list, multipart, organize, organize-polling, poster-cover-reactivity, preview-access, rescrape, update-mode).
- **Flaky tests eliminated** + prettier added to pre-commit.
- **Playwright test helper + spec** migrated `content_id` → `code` to match projection layer field naming.
- **Import guard** added (`scripts/check_import_guard.sh`) enforcing ADR-0020 dependency direction (no `internal/models` → `internal/config` back-edge), wired into `make ci`.
- **Exec-bit regression on `scripts/validate-config-sync.sh`** restored (had broken `make config-drift` / `make ci`).

## Verification

- 97 Go packages PASS under `-race -count=1`
- `gofmt` clean, `go vet` exit 0, `golangci-lint` 0 issues
- `svelte-check`: 0 errors / 0 warnings; `vitest`: 193 tests passed
- `make config-drift` PASS, `make ci` PASS end-to-end (vet lint vuln coverage-check test-race config-drift check-import-guard)
- `make check-swagger`: docs in sync
- v2-scan investigation: all 5 previously-deferred architectural candidates permanently closed (3 STALE, 2 EARNING KEEP)

Three independent reviewer subagents (build/CI, code coherence, merge mechanics) returned consensus **READY** on this branch.

## Merge mechanics

Clean fast-forward: `main` is a strict ancestor of `refactor/architecture-overhaul` (merge-base = `main` tip). No merge commit, no conflict resolution required. 1 commit ahead, 0 behind.

## Summary by CodeRabbit

* **New Features**
  * Real-time job progress in the UI now uses authoritative WebSocket job totals for smoother, more accurate progress.
  * Added Playwright full-stack E2E test entrypoints to expand end-to-end coverage.
* **Bug Fixes**
  * Fixed progress calculation to avoid inflation from terminal per-file states and improved monotonic organize progress updates.
  * Improved review/rescrape UI consistency for movie/poster fields and terminal status handling.
* **Documentation / Dev Notes**
  * Updated configuration examples and dev/testing docs to use `rate_limit` instead of `request_delay`.

### CODE REVIEW

**Comment by seeyabye** [comment]
@copilot perform a code review

**Comment by seeyabye** [comment]
@coderabbitai review

**Comment by seeyabye** [comment]
@coderabbitai full review

**Comment by seeyabye** [comment]
Hey @coderabbitai, I added a `.coderabbit.yaml` with `path_filters` to narrow the review scope from 1500+ files down to 114 production files in the architecturally significant packages (`internal/worker`, `internal/api/{system,core,batch}`, `internal/workflow`).

The path_filters exclude:
- All `*_test.go` files (891 files)
- Generated mocks (`internal/mocks/**`)
- Generated swagger docs (`docs/swagger/**`)
- Frontend (`web/**`)
- Non-critical packages (TUI, scrapers, config, database, models, NFO, organizer, etc.)

The `path_instructions` highlight the 4 files that contain the Copilot review fixes:
1. `internal/worker/job_store_persist.go` — `wireJobDeps` + `reconstructBatchJob` (MovieRepo/BatchCfg/PosterGen restoration)
2. `internal/worker/job_store.go` — `SetReconstructionDeps` (re-hydration of existing jobs)
3. `internal/api/system/proxy.go` — token scope `"direct"` → `"global"` fix
4. `internal/api/core/runtime_manager.go` — `buildBatchJobFactory` wiring of `SetReconstructionDeps`

Could you please re-review the PR with these filters applied?

**Comment by seeyabye** [comment]
Thanks for the review @coderabbitai! I verified all 3 findings and addressed them in commit `c40ef6a7`:

**1. Makefile `.PHONY` (Minor) — ✅ Fixed**
`test-e2e-fullstack` and `test-e2e-field-drop` targets existed (lines 308, 313) but were not declared in `.PHONY`. Added them to the third `.PHONY` line.

**2. `docs/12-testing-guide.md` mock signature (Critical) — ✅ Fixed**
Two issues found:
- **Return-value mismatch** (`_, err` to `_, _, err`): Already fixed in the prior commit `a58e15f1`.
- **`ProgressFunc` type**: The mock used `workflow.ProgressFunc` which doesn't exist — `ProgressFunc` is defined in `internal/scrape/scrape.go`, not `internal/workflow/`. Changed to `scrape.ProgressFunc` in both the `Scrape` and `Apply` mock methods to match the actual `WorkflowInterface` contract.

**3. `go.mod` golang.org/x security (Critical) — ❌ Not actionable**
All recommended versions are already present in `go.mod`:
- `golang.org/x/net` v0.55.0 ✅ (CodeRabbit confirmed "appear correct in the current diff")
- `golang.org/x/sys` v0.45.0 ✅
- `golang.org/x/text` v0.37.0 ✅ (already present as indirect dep)

No changes needed — the dependencies were already at the versions CodeRabbit recommended.

**Comment by seeyabye** [comment]
@coderabbitai please review the latest 3 commits (13b6f9fa, 75bb3dd9, 2c5a9f74) which address subagent review findings across 16 files. The `.coderabbit.yaml` path_filters are still active so the review scope should be narrowed to the 50 production files in `internal/worker`, `internal/api/{system,core,batch}`, and `internal/workflow`.

Key changes to review:

1. **internal/api/system/proxy.go** — Proxy hash normalization: copies `Profile` → `DefaultProfile` before hashing so the test-endpoint token hash matches the save-endpoint validation hash. Also maps token scope `"direct"` → `"global"`.

2. **internal/workflow/apply_orchestrator.go** — Added nil guard for `o.nfo` in `stepMerge` (the only apply step that unconditionally dereferenced its dependency without a nil check).

3. **internal/workflow/scrape_orchestrator.go + compare_orchestrator.go** — Fixed misleading error message ("movieRepo was nil" → "scraper was nil").

4. **internal/worker/job_store_persist.go** — Fixed BatchCfg guard comment + added MovieRepo restoration in `wireJobDeps` + added BatchCfg/PosterGen/Matcher restoration in `reconstructBatchJob`.

5. **internal/testutil/helpers.go** — Consolidated duplicate `UnreachableConfigPath`/`InvalidConfigPath` helpers (alias pattern).

6. **internal/api/auth/storage_miss3_test.go** — Replaced file-level `!windows` build tag with per-test `runtime.GOOS` skips (recovers 11 tests of Windows CI coverage).

7. **Various test files** — Replaced all `/tmp/javtest` hardcoded paths with `t.TempDir()`/`os.TempDir()`.

8. **internal/worker/job_store_reconstruct_test.go** — Removed unnecessary `.Maybe()` mock expectation and unused `testify/mock` import.

**Comment by seeyabye** [comment]
@coderabbitai please review the latest commit f8141110 which addresses the 3 findings from your previous review (RunMigrationsOnStartup context parameter, test-coverage .PHONY, rate_limit config sync). All findings should now be resolved.

**Comment by seeyabye** [comment]
@coderabbitai review

**Comment by seeyabye** [comment]
@coderabbitai full review

**Comment by seeyabye** [comment]
@coderabbitai review

**Comment by seeyabye** [comment]
@coderabbitai full review

File: internal/worker/job_controller.go [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+		c.job.cfg.tempDir = cfg.TempDir
+	}
+	c.job.mu.Unlock()
+
+	ctx, cancel := context.WithCancel(ctx)
+	c.job.lifecycle.setCancelFunc(cancel)
+	if err := c.markStarted(models.JobStatusCompleted); err != nil {
+		cancel()
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (seeyabye) [reply internal/worker/job_controller.go]: Verified — this is intentionally **not** the ordering we use. Moving `setCancelFunc` after `markStarted` would introduce a regression, as confirmed by a dedicated security review:

- **Current (correct) ordering: `setCancelFunc` → `markStarted`** — If a concurrent `Cancel()` arrives between them, it sets the job status to `Cancelled`. Then `markStarted` fails (status ≠ expectedFrom) and the goroutine never starts. The cancellation is honored.

- **Proposed ordering: `markStarted` → `setCancelFunc`** — If `Cancel()` arrives between them, the status is already `Running` but the cancel func isn't stored yet. The new context is never cancelled, so the goroutine starts with an uncancelled context despite the job being marked `Cancelled`. This is the bug.

The comment on line 110 documents this: `"Same ordering as StartScrape: setCancelFunc before markStarted."` Calling `cancel()` on the error path is a safe no-op on an already-cancelled context.

File: internal/workflow/factory.go [hunk 2]
```diff
@@ -0,0 +1,576 @@
...
+	if content.MovieRepo == nil {
+		return nil, nil, fmt.Errorf("buildScraper: movieRepo must not be nil")
+	}
+	agg := aggregator.New(aggCfg,
+		aggregator.NewGenreProcessor(aggCfg.Metadata, replacement.GenreReplacementRepo),
+		aggregator.NewWordProcessor(aggCfg.Metadata, replacement.WordReplacementRepo),
+		aggregator.NewAliasResolver(aggCfg.Metadata, content.ActressAliasRepo),
+	)
+	return scrape.New(registry, agg, content.ActressRepo, content.MovieRepo, httpClient, scrapeCfg, translator, fs), agg, nil
+}
```
[the remark this answers is not in the document]
Comment (seeyabye) [reply internal/workflow/factory.go:218]: Verified — this is intentionally omitted. `scrape.New` already tolerates a nil `actressRepo`:

- **`scrape.go:227`:** `if actressRepo != nil { ... }` — nil-checked before `enrichActressesFromDB`, so nil safely skips actress enrichment.
- **`cache.go:18`:** `if s.movieRepo == nil { return nil, false }` — nil-checked in `tryCache` before `FindByID`.

The aggregator's `GenreProcessor` / `WordProcessor` / `AliasResolver` also handle nil replacement repos by no-oping. This allows partial wiring in setups that only exercise input validation without a full DB (e.g. `TestSecurity_InputValidation` constructs a workflow with nil repos to test request validation paths).

We verified this in an earlier round: adding the hard nil check for `ActressRepo` broke those security tests. The code comment at lines 197-208 documents this reasoning.

**Comment by seeyabye** [comment]
@coderabbitai review

File: internal/worker/fs_case.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	defer func() { _ = fs.Remove(testFile1) }()
+	defer func() { _ = fs.Remove(testFile2) }()
+
+	if err := afero.WriteFile(fs, testFile1, []byte("test"), 0644); err != nil {
+		return false
+	}
+
+	if err := afero.WriteFile(fs, testFile2, []byte("test2"), 0644); err != nil {
+		return false
+	}
```
[the remark this answers is not in the document]
Comment (seeyabye) [reply internal/worker/fs_case.go]: Verified — this is fixed in commit 367cc937. The probe filenames are now randomized via crypto/rand hex tokens (`randomProbeToken()` in fs_case.go), so concurrent probes and pre-existing files are no longer at risk of being truncated. Per-path probes are additionally serialized by `probeLock`. Added `TestRandomProbeToken_DistinctAcrossCalls` to guard uniqueness.

File: internal/workflow/factory.go [hunk 4]
```diff
@@ -0,0 +1,589 @@
...
+// configured. Returns an error if the factory was constructed without Scraper.
+func (f *WorkflowFactory) NewScrapeOnlyWorkflow() (WorkflowInterface, error) {
+	if _, ok := f.cachedScrape.(noOpScrapeOrchestrator); ok {
+		return nil, fmt.Errorf("scrape-only workflow: scraper must not be nil")
+	}
+
+	return &Workflow{
+		scrape:    f.cachedScrape,
+		apply:     noOpApplyOrchestrator{},
+		compare:   f.cachedCompare,
```
[the remark this answers is not in the document]
Comment (seeyabye) [reply internal/workflow/factory.go:534]: Verified — this finding is invalid and intentionally not fixed. `NewScrapeOnlyWorkflow` intentionally allows a `noOpCompareOrchestrator` when NFO is not configured, because the scrape-only workflow does not invoke `Compare` — it only scrapes. `cachedCompare` becomes noOp only when `fc.NFOIface == nil` (factory.go:455), which is a valid configuration for `javinizer scrape` when the user has not set up NFO. Adding the suggested validation would break the scrape CLI for users without NFO configured, since `NewScrapeOnlyWorkflow` would then reject a factory built without NFOIface. The scrape orchestrator (`scrape_orchestrator.go`) never references compare, so the noOp is safe. If `Compare` were ever called on such a workflow, `noOpCompareOrchestrator.Execute` returns a clear "compare not configured" error — failing loudly, not silently.

**Comment by seeyabye** [comment]
@coderabbitai review

**Comment by seeyabye** [comment]
@coderabbitai review

Could you review the two most recent commits on this PR?

- `af63138b` — cycle-2/3 review fallout: a source-named out-of-range rating warning (relocates a dead `assignRating` block into `getRatingByPriorityWithSource`'s skip path, also restoring the source name the old warning lost); `>100%` caps on both the `!isRunning` and `isRunning`+no-messages branches of `computeJobProgress`; and corrected stale `ShouldCropPoster` "never empty" test comments under the `!v` predicate.
- `8bb1a89e` — cycle-3 residuals: a `Prepare` guard that surfaces an explicitly empty `scrapers.priority` (`priority: []`/`null`) as a config error (prevents a silent blank-movie data-loss path), plus a `goconst` cleanup extracting a `translationProviderOpenAI` constant.

Context: these came out of a multi-cycle regression-review loop. The full gate is green — go build/vet/gofmt, golangci-lint 0 issues, `go test -race -count=1 ./...` 97/97 ok, svelte-check 0 errors, vitest 117 pass. I'm most interested in any regressions or edge cases we missed, especially around the `Prepare`-vs-`Validate` layering of the priority guard (it's intentionally in `Prepare` so pure-validation tests building minimal configs aren't forced to set a priority).

**Comment by seeyabye** [comment]
@coderabbitai review

Please do a thorough review of the recent WebSocket-driven progress-bar changes (pushed today, not yet reviewed). These are the commits in scope:

- `864e8110` — feat(web): monotonic+verbose WS-driven progress bars (Home bar, organize file names, WS totals)
- `debd395c` — fix(web): gate organize bar-drive to aggregate messages (F-1 review-page bar oscillation)
- `d86eec39` — test(e2e): filter monotonic-delivery check to aggregate pending (no file_path)

Focus areas (production code only; tests are excluded by path_filters):
- **Backend broadcast:** `internal/websocket/hub.go`, `internal/worker/apply_phase.go`, `internal/worker/batch_job_interface.go`, `internal/api/batch/apply_config_builder.go`, `internal/api/batch/rescrape_orchestrator.go` — the per-file organize progress broadcast + the aggregate high-water-mark mutex.
- **Frontend:** `web/frontend/src/routes/review/[jobId]/logic/organize-controller.ts`, `web/frontend/src/routes/+page.svelte`, `web/frontend/src/lib/components/ProgressModal.svelte`, `web/frontend/src/lib/components/BackgroundJobIndicator.svelte`, `web/frontend/src/lib/api/types.ts` — monotonic bar driving, per-file start frames vs aggregate frames, WS totals.

Key invariant to verify: the organize bar is driven ONLY by aggregate progress frames (`!file_path`); per-file start frames (`file_path` set, Progress:0) are display-only and must not regress the bar. Leave no stone unturned — correctness of the monotonic high-water mark, any race/toctou in the broadcast hub, and frontend clamping at 100%.

**Comment by seeyabye** [comment]
@coderabbitai review

Rebased `refactor/architecture-overhaul` onto `main` (400d284e) and force-pushed. Also updated `.coderabbit.yaml`: added frontend test exclusions (`*.test.ts`, `*.spec.ts`, `web/frontend/tests/**`) for parity with the existing Go `*_test.go` exclusion, and documented the 2-pass chunked review plan.

After filtering, ~187 in-scope files remain (over the 150-file/review limit), split as:
- **Pass 1 — backend** (~117): `internal/worker/`, `internal/workflow/`, `internal/websocket/`, `internal/api/{batch,core,system}/`
- **Pass 2 — frontend + config** (~70): `web/frontend/` source + `.github/`, `Makefile`, `configs/`, `docs/`

Please review incrementally (unreviewed files first). Start with the backend chunk, then the frontend chunk.

**Comment by seeyabye** [comment]
@coderabbitai full review

Pass 1 of the chunked full review. `.coderabbit.yaml` is now scoped to **backend only** (~117 files): `internal/worker/`, `internal/workflow/`, `internal/websocket/`, `internal/api/{batch,core,system}/`. Frontend source + config/docs are temporarily excluded and will be Pass 2.

Please do a **full** review (not incremental) of the backend changes — the workflow seam (ScrapePhase/ApplyPhase/RescrapePhase, JobRunner), job store reconstruction (SetReconstructionDeps/wireJobDeps), the WS progress broadcast hub (high-water-mark mutex, aggregate vs per-file frames), and the batch apply/rescrape paths. Focus on data-integrity, panic safety, and concurrency correctness.

**Comment by seeyabye** [comment]
@coderabbitai full review

Pass 2 of the chunked full review. `.coderabbit.yaml` is now scoped to **frontend source + config/docs** (~70 files): `web/frontend/` source (tests excluded), `.github/workflows/`, `Makefile`, `configs/`, `docs/`, `scripts/`, `go.mod`, `go.sum`. All `internal/**` backend is excluded — Pass 1 already full-reviewed it (19 actionable findings, incl. a Critical `applyFile` unnamed-return panic-recovery bug).

Please do a **full** review (not incremental) of the frontend WS-driven progress-bar feature (`web/frontend/src/lib/utils/job-progress.ts` terminal-status set + `computeJobProgress` clamping, `web/frontend/src/routes/review/[jobId]/logic/organize-controller.ts` monotonic bar logic, the ProgressModal/BackgroundJobIndicator consumers) and the config/docs/scripts. Focus on stuck-counter/premature-terminal regressions, bar-overflow, and WS-contract drift.

**Comment by seeyabye** [comment]
@coderabbitai full review

Triggering a **full review** (bypass incremental cache) for Pass 1 of a 5-pass chunked full review. This pass is scoped to `internal/api/**` (137 files) via `.coderabbit.yaml` path_filters. The other 4 passes will follow on subsequent commits; after all land, the config is restored to full scope.

**Comment by seeyabye** [comment]
@coderabbitai full review

Re-triggering Pass 1 full review. The `.coderabbit.yaml` now uses negative-exclude path_filters (proven approach) scoped to `internal/api/**` (137 files). Previous positive-only attempt no-op'd.

**Comment by seeyabye** [comment]
@coderabbitai full review

This is **Pass 2 of 5** of the chunked full review. The `.coderabbit.yaml` on this commit (`a7e93255`) scopes the review to **orchestration + config** via negative-excludes:

- `internal/{worker,workflow,websocket,config}/**`
- `cmd/**`, `.github/**`, `docs/**`
- root: `Makefile`, `go.mod`

**115 in-scope files** (under the 150 limit). `internal/api/**` (Pass 1) and the other passes are excluded so this pass stays within limits.

Pass 1 (`internal/api`, 137 files) is **complete** — review landed and all 20 valid findings were fixed in `10d05f43`.

Please run a full review for Pass 2 only, using the updated path_filters/path_instructions in `.coderabbit.yaml`.

**Comment by seeyabye** [comment]
@coderabbitai full review

This is **Pass 3 of 5** of the chunked full review. The `.coderabbit.yaml` on this commit (`481e7cfc`) scopes the review to **scrapers + scrape infrastructure** via negative-excludes:

- `internal/{scraper,scraperconfig,scraperutil,scrape,ssrf,httpclient,ratelimit,challengedetect}/**`

**91 in-scope files** (under the 150 limit). All other passes are excluded so this pass stays within limits.

Passes 1 (`internal/api`, 137 files) and 2 (orchestration+config, 115 files) are **complete** — reviews landed and all findings fixed (`10d05f43`, `ea6928a9`).

Please run a full review for Pass 3 only, using the updated path_filters/path_instructions in `.coderabbit.yaml`.

**Comment by seeyabye** [comment]
@coderabbitai full review

This is **Pass 4 of 5** of the chunked full review. The `.coderabbit.yaml` on this commit (`3233f985`) scopes the review to **data + processing** via negative-excludes:

- `internal/{database,organizer,downloader,nfo,template,translation,history,aggregator,matcher,scanner,eventlog,fsutil,imageutil,poster,mediainfo,formatter}/**`

**122 in-scope files** (under the 150 limit). All other passes are excluded so this pass stays within limits.

Passes 1 (`internal/api`, 137 files), 2 (orchestration+config, 115 files), and 3 (scrapers+scrape-infra, 91 files) are **complete** — reviews landed and all findings fixed (`10d05f43`, `ea6928a9`, `d4f48d1f`).

Please run a full review for Pass 4 only, using the updated path_filters/path_instructions in `.coderabbit.yaml`.

**Comment by seeyabye** [comment]
@coderabbitai full review

Re-pinging — the prior full-review request for Pass 4 appears not to have been processed (no ack and no review landed after ~3h). This is **Pass 4 of 5** of the chunked full review.

The `.coderabbit.yaml` on this commit (`3233f985`) scopes the review to **data + processing** via negative-excludes:

- `internal/{database,organizer,downloader,nfo,template,translation,history,aggregator,matcher,scanner,eventlog,fsutil,imageutil,poster,mediainfo,formatter}/**`

**122 in-scope files** (under the 150 limit). All other passes are excluded so this pass stays within limits.

Passes 1 (`internal/api`, 137 files), 2 (orchestration+config, 115 files), and 3 (scrapers+scrape-infra, 91 files) are **complete** — reviews landed and all findings fixed (`10d05f43`, `ea6928a9`, `d4f48d1f`).

Please run a full review for Pass 4 only, using the updated path_filters/path_instructions in `.coderabbit.yaml`.

**Comment by seeyabye** [comment]
@coderabbitai full review

This is **Pass 5 of 5** (the final pass) of the chunked full review.

The `.coderabbit.yaml` on this commit (`d38b37f5`) scopes the review to **frontend + tui + utils** via negative-excludes:

- `web/frontend/**` (54 files) — SvelteKit SPA, API clients, stores, review flow
- `internal/tui/**` (36 files) — Bubble Tea model/view/handlers
- `internal/models/**` (20 files) — domain models + GORM tags
- `internal/{commandutil,configutil,update,testutil,version,panicutil,operationmode,logging,enumutil,coverage}/**` (22 files) — utils + DI wiring

**132 in-scope files** (under the 150 limit). All other passes are excluded so this pass stays within limits.

Passes 1 (`internal/api`), 2 (orchestration+config), 3 (scrapers+scrape-infra), and 4 (data+processing) are **complete** — reviews landed and all findings fixed.

Please run a full review for Pass 5 only, using the updated path_filters/path_instructions in `.coderabbit.yaml`. This is the final pass; once complete I'll restore `.coderabbit.yaml` to full scope.

File: internal/update/service.go [hunk 5]
```diff
@@ -6,82 +6,85 @@ import (
...
+			Source: UpdateSourceNone,
 		}, nil
 	}
 
 	// If state is stale and we should check, do it in background
 	if s.store.ShouldCheck() {
-		go s.BackgroundCheck()
+		// Use Background() so the check outlives the request — intentional, not a leak
+		go s.BackgroundCheck(context.Background())
 	}
```
[the remark this answers is not in the document]
Comment (seeyabye) [reply internal/update/service.go]: Not fixing in this PR — leaving the nil-state skip as-is and tracking the proper fix in a follow-up issue.

**Why the suggested fix was reverted:** firing `go s.BackgroundCheck(...)` in the nil branch turns `GetStatus` into a side-effectful path. `printUpdateStatus` (`cmd/javinizer/commands/info/command.go:113`) calls `GetStatus` synchronously for the read-only `javinizer info` command, so a first run with no cache would fire a real network call to GitHub + write `data/update_cache.json`. This broke two tests when attempted:
- `TestPrintUpdateStatus_NeverChecked` — the bg goroutine wrote a real cache the test then read back.
- `TestRunInfo_WriteFailures_Exhaustive` — the added `SaveState` write changed the write count the test measures.

So the "quick win" is actually a behavior change to a read-only command (non-deterministic network + disk write), which is why the existing skip comment at `service.go:71-76` exists.

**The bigger picture (investigated while evaluating this):** the nil-state skip is the *smallest* of three compounding gaps. The dominant reasons a fresh install never learns about updates:
1. `StartBackgroundCheck` (the periodic ticker, `service.go:176`) is fully implemented and tested but **never wired in production** — zero callers outside `_test.go`. Not the CLI root, not the API server.
2. No check at startup — `PersistentPreRun` (`root.go:59`) only runs `initConfig()`.
3. The nil-state skip (this thread).

All three compound: no boot check, no periodic check, and the read path won't backfill the cache → users only learn about updates by manually hitting the force-check endpoint.

**Proper fix (follow-up issue):** address all three together — wire `StartBackgroundCheck` at API server start (gated on `VersionCheckEnabled`, stopped on shutdown), add a one-shot `BackgroundCheck` at startup, and add a checker-injection seam to `NewService` so these can be tested hermetically. `GetStatus` stays a pure read; the nil-state skip becomes benign once the cache is populated at startup/periodically. Tracked locally as `ready-for-human`.

[the remark this answers is not in the document]
Comment (seeyabye) [reply internal/update/service.go]: @coderabbitai please open a Github issue
